### ISSUE TITLE
BUG: EDS element line markers do not scale correctly

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#1961]
Noticed on latest Release (1.3.1):

![eds_plotting_issue](https://user-images.githubusercontent.com/1278301/40491761-e7e0a728-5f3c-11e8-9039-78edd79660f3.gif)

EDS element line markers work correctly in the first pixel of a plotted spectrum image, but do not adjust correctly when moving around the navigator. This appears to be working in the bundle version of 1.3. I am currently downloading the latest bundle release to see if it is an issue there as well.

Details of my environment:

```
[hyperspy] λ conda list
# packages in environment at C:\anaconda3\envs\hyperspy:
#
alabaster                 0.7.10                   py36_1    conda-forge
asn1crypto                0.24.0                   py36_0    conda-forge
astroid                   1.6.3                    py36_0    conda-forge
attrs                     17.4.0                     py_0    conda-forge
autopep8                  1.3.4                     <pip>
babel                     2.5.3                    py36_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
backports                 1.0                      py36_1    conda-forge
backports.functools_lru_cache 1.5                      py36_0    conda-forge
blas                      1.0                         mkl
bleach                    2.1.3                      py_0    conda-forge
bokeh                     0.12.15                  py36_0    conda-forge
ca-certificates           2018.1.18                     0    conda-forge
certifi                   2018.4.16                 <pip>
certifi                   2018.1.18                py36_0    conda-forge
cffi                      1.11.5                   py36_0    conda-forge
chardet                   3.0.4                    py36_0    conda-forge
click                     6.7                        py_1    conda-forge
cloudpickle               0.5.2                      py_0    conda-forge
colorama                  0.3.9                    py36_0    conda-forge
cryptography              2.2.1                    py36_0    conda-forge
cssselect                 1.0.3                     <pip>
cycler                    0.10.0                   py36_0    conda-forge
cytoolz                   0.9.0.1                  py36_0    conda-forge
dask                      0.17.2                     py_0    conda-forge
dask-core                 0.17.2                     py_0    conda-forge
decorator                 4.3.0                     <pip>
decorator                 4.2.1                    py36_0    conda-forge
dill                      0.2.7.1                  py36_0    conda-forge
distributed               1.21.6                   py36_0    conda-forge
docutils                  0.14                     py36_0    conda-forge
entrypoints               0.2.3                    py36_1    conda-forge
factor-analyzer           0.2.2                     <pip>
fastcache                 1.0.2                    py36_0    conda-forge
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
future                    0.16.0                    <pip>
gitdb2                    2.0.3                     <pip>
GitPython                 2.1.9                     <pip>
h5py                      2.7.1                    py36_2    conda-forge
hdf5                      1.10.1                   vc14_2  [vc14]  conda-forge
heapdict                  1.0.0                    py36_0    conda-forge
howdoi                    1.1.13                    <pip>
html5lib                  1.0.1                      py_0    conda-forge
hyperspy-gui-traitsui     1.0.3                     <pip>
icc_rt                    2017.0.4             h97af966_0
icu                       58.2                     vc14_0  [vc14]  conda-forge
idna                      2.6                      py36_1    conda-forge
imageio                   2.3.0                    py36_0    conda-forge
imagesize                 1.0.0                    py36_0    conda-forge
intel-openmp              2018.0.0                      8
ipykernel                 4.8.2                    py36_0    conda-forge
ipympl                    0.1.1                     <pip>
ipyparallel               6.1.1                     <pip>
ipython                   6.3.1                    py36_0    conda-forge
ipython_genutils          0.2.0                    py36_0    conda-forge
ipywidgets                7.2.1                    py36_1    conda-forge
isort                     4.3.4                    py36_0    conda-forge
jedi                      0.12.0                    <pip>
jedi                      0.11.1                   py36_0    conda-forge
jinja2                    2.10                     py36_0    conda-forge
joblib                    0.11                     py36_0    conda-forge
jpeg                      9b                       vc14_2  [vc14]  conda-forge
jsonschema                2.6.0                    py36_1    conda-forge
jupyter                   1.0.0                      py_1    conda-forge
jupyter_client            5.2.3                    py36_0    conda-forge
jupyter_console           5.2.0                    py36_0    conda-forge
jupyter_contrib_core      0.3.3                    py36_1    conda-forge
jupyter_contrib_nbextensions 0.5.0                    py36_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyter_highlight_selected_word 0.2.0                    py36_0    conda-forge
jupyter_latex_envs        1.4.4                    py36_0    conda-forge
jupyter_nbextensions_configurator 0.4.0                    py36_0    conda-forge
jupyternotify             0.1.15                    <pip>
kiwisolver                1.0.1                    py36_1    conda-forge
lazy-object-proxy         1.3.1                    py36_0    conda-forge
lesscpy                   0.13.0                   py36_0    conda-forge
libflang                  5.0.0             vc14_20180207  [vc14]  conda-forge
libiconv                  1.14                     vc14_4  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libsodium                 1.0.16                   vc14_0  [vc14]  conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
libxml2                   2.9.5                    vc14_1  [vc14]  conda-forge
libxslt                   1.1.32                   vc14_0  [vc14]  conda-forge
link-traits               1.0.2                    py36_0    conda-forge
llvm-meta                 6.0.0                         0    conda-forge
llvmlite                  0.21.0                   py36_0    conda-forge
locket                    0.2.0                    py36_1    conda-forge
lxml                      4.2.1                    py36_0    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
markupsafe                1.0                      py36_0    conda-forge
matlabengineforpython     R2018a                    <pip>
matplotlib                2.2.2                    py36_1    conda-forge
mccabe                    0.6.1                    py36_0    conda-forge
mistune                   0.8.3                      py_0    conda-forge
mkl                       2018.0.2                      1
mkl_fft                   1.0.2                    py36_0    conda-forge
mkl_random                1.0.1                    py36_0    conda-forge
more-itertools            4.1.0                      py_0    conda-forge
mpmath                    1.0.0                      py_0    conda-forge
mrcz                      0.3.6                     <pip>
msgpack-python            0.5.6                    py36_0    conda-forge
msys2-conda-epoch         20160418                      1
multiprocess              0.70.5                   py36_0    conda-forge
natsort                   5.2.0                      py_0    conda-forge
nbconvert                 5.3.1                      py_1    conda-forge
nbdime                    0.4.1                     <pip>
nbformat                  4.4.0                    py36_0    conda-forge
networkx                  2.1                      py36_0    conda-forge
nodejs                    9.11.0                        0    conda-forge
nose                      1.3.7                    py36_2    conda-forge
notebook                  5.4.1                    py36_0    conda-forge
numba                     0.36.2          np114py36h12cb543_0
numpy                     1.14.3           py36h9fa60d3_1
numpy-base                1.14.3           py36h555522e_1
numpydoc                  0.8.0                    py36_0    conda-forge
olefile                   0.45.1                   py36_0    conda-forge
openblas                  0.2.20                   vc14_7  [vc14]  conda-forge
openmp                    6.0.0                    vc14_0  [vc14]  conda-forge
openssl                   1.0.2n                   vc14_0  [vc14]  conda-forge
packaging                 17.1                       py_0    conda-forge
pandas                    0.22.0                   py36_0    conda-forge
pandoc                    2.1.3                         0    conda-forge
pandocfilters             1.4.2                     <pip>
pandocfilters             1.4.1                    py36_0    conda-forge
parso                     0.1.1                      py_0    conda-forge
parso                     0.2.0                     <pip>
partd                     0.3.8                    py36_0    conda-forge
pathos                    0.2.1                    py36_1    conda-forge
patsy                     0.5.0                    py36_0    conda-forge
pep8                      1.7.1                     <pip>
pickleshare               0.7.4                    py36_0    conda-forge
pillow                    5.1.0                    py36_0    conda-forge
pint                      0.8.1                    py36_0    conda-forge
pip                       9.0.3                     <pip>
pip                       9.0.3                    py36_0    conda-forge
pluggy                    0.6.0                      py_0    conda-forge
pluggy                    0.6.0                     <pip>
ply                       3.11                     py36_0    conda-forge
pockets                   0.5.1                     <pip>
pox                       0.2.3                    py36_0    conda-forge
ppft                      1.6.4.7.1                py36_0    conda-forge
prompt_toolkit            1.0.15                   py36_0    conda-forge
psutil                    5.4.3                    py36_0    conda-forge
py                        1.5.2                     <pip>
py                        1.5.3                      py_0    conda-forge
pycodestyle               2.3.1                     <pip>
pycodestyle               2.4.0                    py36_0    conda-forge
pycparser                 2.18                     py36_0    conda-forge
pyface                    6.0.0                     <pip>
pyflakes                  1.6.0                     <pip>
pyflakes                  1.6.0                    py36_0    conda-forge
pygments                  2.2.0                    py36_0    conda-forge
pylint                    1.8.4                    py36_0    conda-forge
pyopenssl                 17.5.0                   py36_1    conda-forge
pyparsing                 2.2.0                    py36_0    conda-forge
pypiwin32                 223                       <pip>
pyqode.core               2.11.0                    <pip>
pyqode.python             2.10.0                    <pip>
pyqode.qt                 2.10.0                    <pip>
pyqt                      5.6.0                    py36_4    conda-forge
pyquery                   1.4.0                     <pip>
pyreadline                2.1              py36h094d952_1
pysocks                   1.6.8                    py36_1    conda-forge
pytest                    3.5.0                    py36_0    conda-forge
pytest                    3.3.1                     <pip>
pytest-mpl                0.9                        py_0    conda-forge
python                    3.6.5                         1    conda-forge
python-dateutil           2.7.2                      py_0    conda-forge
python-qt-binding         0.2.17                    <pip>
pytz                      2018.4                     py_0    conda-forge
pywavelets                0.5.2                    py36_1    conda-forge
pywin32                   223                       <pip>
pywinpty                  0.5                      py36_2    conda-forge
pyyaml                    3.12                     py36_1    conda-forge
pyzmq                     17.0.0                   py36_4    conda-forge
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
qtawesome                 0.4.4                    py36_0    conda-forge
qtconsole                 4.3.1                    py36_0    conda-forge
qtpy                      1.4.0                    py36_0    conda-forge
requests                  2.18.4                   py36_1    conda-forge
requests-cache            0.4.13                    <pip>
rope                      0.10.7                   py36_0    conda-forge
scikit-image              0.13.1                   py36_0    conda-forge
scikit-learn              0.19.1           py36h53aea1b_0
scipy                     1.0.1            py36hce232c7_0
send2trash                1.5.0                      py_0    conda-forge
setuptools                39.1.0                    <pip>
setuptools                39.0.1                   py36_0    conda-forge
simplegeneric             0.8.1                    py36_0    conda-forge
sip                       4.18                     py36_1    conda-forge
six                       1.11.0                   py36_1    conda-forge
smmap2                    2.0.3                     <pip>
snowballstemmer           1.2.1                    py36_0    conda-forge
sortedcontainers          1.5.9                    py36_0    conda-forge
spectral                  0.19                      <pip>
sphinx                    1.7.2                    py36_0    conda-forge
sphinx_rtd_theme          0.3.0                    py36_0    conda-forge
sphinxcontrib-napoleon    0.6.1                     <pip>
sphinxcontrib-newsfeed    0.1.4                     <pip>
sphinxcontrib-websupport  1.0.1                    py36_0    conda-forge
sqlite                    3.22.0                   vc14_0  [vc14]  conda-forge
statsmodels               0.8.0                    py36_0    conda-forge
sympy                     1.1.1                    py36_0    conda-forge
tblib                     1.3.2                    py36_0    conda-forge
terminado                 0.8.1                    py36_0    conda-forge
testpath                  0.3.1                    py36_0    conda-forge
tk                        8.6.7                    vc14_0  [vc14]  conda-forge
toolz                     0.9.0                      py_0    conda-forge
tornado                   5.0.2                    py36_0    conda-forge
tqdm                      4.21.0                     py_0    conda-forge
traitlets                 4.3.2                    py36_0    conda-forge
traits                    4.6.0                    py36_1    conda-forge
traitsui                  6.0.0                     <pip>
typing                    3.6.4                    py36_0    conda-forge
urllib3                   1.22                     py36_0    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.0.25420                    0    conda-forge
wcwidth                   0.1.7                    py36_0    conda-forge
webencodings              0.5.1                     <pip>
webencodings              0.5                      py36_0    conda-forge
wheel                     0.31.0                   py36_0    conda-forge
widgetsnbextension        3.2.1                    py36_0    conda-forge
win_inet_pton             1.0.1                    py36_1    conda-forge
wincertstore              0.2                      py36_0    conda-forge
winpty                    0.4.3                    vc14_2  [vc14]  conda-forge
wrapt                     1.10.11                  py36_0    conda-forge
yaml                      0.1.7                    vc14_0  [vc14]  conda-forge
zeromq                    4.2.5                    vc14_1  [vc14]  conda-forge
zict                      0.1.3                      py_0    conda-forge
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```

**Comment by jat255** [comment]
Confirming same buggy behavior on [latest bundle](https://github.com/hyperspy/hyperspy-bundle/releases/tag/v2018.04.22-beta1)

**Comment by jat255** [comment]
It appears initially that the issue is related to [`drawing.signal1d.update()`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/drawing/signal1d.py#L137) not being called when the navigation index is changed...

**Comment by jat255** [comment]
So (*I think*) the issue was introduced with commit https://github.com/hyperspy/hyperspy/commit/5d12f523385483e5cba52ae9ada4ba85c6f136db with the removal of the lines

```python
self.axes_manager.events.indices_changed.connect(self.update, [])
self.events.closed.connect(
  lambda: self.axes_manager.events.indices_changed.disconnect(
      self.update), [])
```

If I add those lines back in, the EDS marker lines respond as-expected. I'm not overly familiar with this part of the code though, and I would feel in over my head futzing around with it. Any thoughts, @francisco-dlp and @ericpre ?

**Comment by ericpre** [comment]
Minimum example to reproduce the issue:
``` python
import hyperspy.api as hs

a = hs.datasets.example_signals.EDS_SEM_Spectrum()
s = hs.stack([a, a*5])
s.plot(True)
```

**Comment by ericpre** [comment]
Fixed in #1970.
